src/app-layer-parser.c

```diff
@@ -655,7 +662,7 @@ uint64_t AppLayerTransactionGetActiveLogOnly(Flow *f, uint8_t flags)
     AppLayerParserProtoCtx *p = &alp_ctx.ctxs[f->protomap][f->alproto];
 
     if (p->logger == TRUE) {
```

Comment:
Currently yes.  They will be updated as each direction reaches completion, whether or not anything was logged (as thats up to the logging callback, but the transition is based on readyness). Then we move forward to the next one. So you are getting the same behaviour as before which waits for both to become ready before anything is logged.

So it should behave exactly the same as now.  Unless I'm missing some special case on the current setup?